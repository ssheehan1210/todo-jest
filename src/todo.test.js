import Todo from './todo';
import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure } from 'enzyme';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

configure ({ adapter: new Adapter() });

test('Todo component renders the text of the todo', () => {
	const todo = { id: 1, done: false, name: 'Finish final project' };
	const wrapper = mount(
		<Todo todo={todo} />
	);

	const p = wrapper.find('.toggle-todo');
	console.log(p.text());
	expect(p.text()).toBe('Finish final project');
})

test('Todo calls doneChange when todo is clicked', () => {
	const doneChange = jest.fn()
	const todo = { id: 1, done: false, name: 'Take out trash' };
	const wrapper = mount(
		<Todo todo={todo} doneChange={doneChange} />
	);

	const p = wrapper.find('.toggle-todo');

	p.simulate('click');

	expect(doneChange).toBeCalledWith(1);
})

describe('Todo component renders the todo correctly', () => {
	it('renders correctly', () => {
		const todo = { id: 1, done: false, name: 'Take out trash' };
		const rendered = renderer.create(
			<Todo todo={todo} />
		);

		expect(rendered.toJSON()).toMatchSnapshot();
	})
})
