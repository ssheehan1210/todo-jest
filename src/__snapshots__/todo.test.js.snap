// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component renders the todo correctly renders correctly 1`] = `
<div
  className="todo  todo-1"
>
  <p
    className="toggle-todo"
    onClick={[Function]}
  >
    Take out trash
  </p>
  <a
    className="delete-todo"
    href="#"
    onClick={[Function]}
  >
    Delete
  </a>
</div>
`;
